File: 02_march_2024_Minimum_absolute_difference_in_BST.py
# class Node:
#     def __init__(self):
#         self.data = None
#         self.left = None
#         self.right = None
import sys         
class Solution:
    def absolute_diff(self, root):
        self.res = sys.maxsize
        self.prev = 0
        self.find(root)
        return self.res

    def find(self, root):
        if not root:
            return

        self.find(root.left)
        if self.prev > 0:
            self.res = min(self.res, abs(root.data - self.prev))
        self.prev = root.data
        self.find(root.right)

File: 02_march_2024_Minimum_absolute_difference_in_BST.cpp
/*The Node structure is defined as
struct Node {
    int data;
    Node *left;
    Node *right;

};
*/

class Solution
{
    public:
    int res = INT_MAX , prev =0;
    
    void find(Node* root ){
        if(root==nullptr) return;
        
        find(root->left);
        if(prev>0) res = min(res , abs(root->data-prev));
        prev = root->data;
        find(root->right);
        
    }
    int absolute_diff(Node *root)
    {
        //Your code here
       
        find(root);
        
        return res ;
    }
};
